### ISSUE TITLE
Feature Request: Utilize KeePass "History" feature

### ISSUE THREAD

**Issue by NJK-HOME** [JustinGrote/SecretManagement.KeePass#37]
I would like to be able to use Set-Secret to update an existing record in KeePass and add the original value to the history list of that record.  Currently, you have to remove the record from Keepass and then re-add it with the same name and new secret any time you want to change it.  If you have a recycle bin setup, it will put the original record there...but I think the history feature is much nicer than having to look through the recycle bin especially when you are in a work-flow that updates 100s of records on a periodic basis.

**Comment by JustinGrote** [comment]
@NJK-HOME I'll put this on the backlog, it should be doable but it will take a bit of rearchitecture of how the underlying PoshKeePass works. My focus for the first few releases was on the reading of secrets, I can delve more into the writing of secrets soon :)

**Comment by NJK-HOME** [comment]
Sounds good.   Thanks for considering it.

**Comment by Callidus2000** [comment]
Hi @NJK-HOME 
Should work now in the freshly released 0.9.2.

**Comment by NJK-HOME** [comment]
Yep - looks good so far.  I used Set-Secret on an item and it created a history entry for it.  I used Remove-Secret on another item and it moved it to the recycle bin.  Perfect!  I will do some more testing, but I think this may be what I was looking for.

Excellent work!
Thank
NJK

**Comment by NJK-HOME** [comment]
I did one more test - created a new item with the same name as one in the recycle bin and it created that new item.  I am pretty sure this is the desired outcome - at least for me.  Maybe others might not want a new item created if it already existed previously, was deleted, and now is in the recycle bin.  But for now, I think that is correct.

Thanks
NJK

**Comment by NJK-HOME** [comment]
OK - I think I might have found an issue.  When using Set-Secret when there are two items with the same name, one being in the recycle bin, it asks to narrow down which one to update.  If I am not using folders, and have everything in the root, how do I specify a root item?

![image](https://user-images.githubusercontent.com/61813693/130536904-c65b4a13-c84e-4d35-88c3-5dfbfab1784a.png)

Let me know if you want me to create a new issue since this is more related to the recycle bin and not history.
NJK

**Comment by NJK-HOME** [comment]
Hi @Callidus2000 and @JustinGrote 
I tested the new beta release and it looks good now.  Thank you so much.  I think this issue can probably be closed if you want.

Thanks,
NJK-HOME
